### PR TITLE
Issue/4322 notifications empty state

### PR DESCRIPTION
**Pull request by elibud**
**Fixes** #4322 

**To test:**
1. Login with an account with no sites to check how the `WPNoResultsView` with an image looks.
2. Create a new site on that account and check the 5 different tabs on Notifications.
emptysitesemptyunread

 

Needs review: @jleandroperez

### CODE REVIEW

**Comment by jleandroperez** [comment]
Is this one ready for review @elibud ?

**Comment by jleandroperez** [comment]
(I know, i know, you've removed the label, just confirming!) 😬

**Comment by elibud** [comment]
@jleandroperez yep, thanks

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift [hunk 1]
```diff
@@ -1026,32 +1026,74 @@ private extension NotificationsViewController {
...
+            .comment: NSLocalizedString("View Reader",
+                                        comment: "Displayed in the Notifications Tab as a button title, when there are no notifications"),
+            .follow: NSLocalizedString("View Reader",
+                                       comment: "Displayed in the Notifications Tab as a button title, when there are no notifications"),
+            .like: NSLocalizedString("View Reader",
+                                     comment: "Displayed in the Notifications Tab as a button title, when there are no notifications")
+        ]
+
+        let filter = Filter(rawValue: filtersSegmentedControl.selectedSegmentIndex) ?? .none
+        return buttonTitleMap[filter] ?? String()
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift]: Note: (by all means, **no need to change anything**!!) I meant to convert this `map` mechanism, to actually, a `Filter property`. Main reason is... suppose:

```
enum Filter {
   //..
   var title: String {
      switch self { 
         case .none: return ...
      }

   }
}
```

By doing something like this, (if) a new filter gets added, we get free compiler warnings for missing case(s). Whereas... this old `map` mechanism keeps on building, and we need to figure out on our own all of the spots that must be updated.

Props to @frosty for that tip, i've adopted that practise a couple months ago... but couldn't get to this refactor!

**Comment by elibud** [comment]
@jleandroperez thank you sir, you are very kind :) 
I've made the suggested changes, can you please take another look?

**Pull request merged**